### ISSUE TITLE
Bug: one-var rule with "never" doesn't detect 2 declarations inside a for loop

### ISSUE THREAD

**Issue by eyal1990** [eslint/eslint#15592]
### Environment

Node version: 14.7.6
npm version: 6.14.8
Local ESLint version: 8.5.0
Global ESLint version: 8.5.0
Operating System: Win 10

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
'one-var': ['error', 'never']
```
</details>

```js
for(let i = 0, j = 2; i < 0; i++) {
    console.log(i, j);
}
```
I could not use the ESLint Demo because it doesn't support adding options which are crucial here ("never" option)

### What did you expect to happen?

Expected an error to appear because 2 variables are defined inside the for loop.

### What actually happened?

No error appears

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
it seems expected, per the test:https://github.com/eslint/eslint/blob/2dc38aa653f1d5137a9abf82024c67a11620bb7c/tests/lib/rules/one-var.js#L34-L37

**Comment by eyal1990** [comment]
> it seems expected, per the test:
> 
> https://github.com/eslint/eslint/blob/2dc38aa653f1d5137a9abf82024c67a11620bb7c/tests/lib/rules/one-var.js#L34-L37

Thank you for the reference.
How is it different from the more usual case that represented in the test here:
https://github.com/eslint/eslint/blob/2dc38aa653f1d5137a9abf82024c67a11620bb7c/tests/lib/rules/one-var.js#L635-L645

I think it should behave the same and not allow multiple declaration inside a for loop.
If there's a solid reason to treat for loops differently it might be better to present another option of the "never" so it will be stricter (and I believe more consistent).

**Comment by aladdin-add** [comment]
just had a little more research: it was introduced in https://github.com/eslint/eslint/issues/2505, while I did not see a full discussion.

I guess the reason: no way to separate the decl to be the same functionality. e.g.
```js
for (var i = 0, len = arr.length; i < len; i++) {}
```

you may want to rewrite to:
```js
var i = 0, len = arr.length;
for (; i < len; i++) {}
```
but `i` and `len` were define in the outer scope.

**Comment by eyal1990** [comment]
#2505 used `"one-var": [2, { initialized: "never" }]`.

It means he should have wrote:
```
var i = 0;
var len = arr.length;
for (; i < len; i++) {}
```

or probably:
```
var len = arr.length;
for (var i = 0; i < len; i++) {}
```

It's completely in line with this rule, while ignoring for loops is not.

**Comment by mdjermanovic** [comment]
> ```js
> for(let i = 0, j = 2; i < 0; i++) {
>     console.log(i, j);
> }
> ```

I agree with @aladdin-add that this works as intended. This is a stylistic rule that enforces `let i = 0; let j = 2;` instead of `let i = 0, j = 2;` where they are equivalent. In this case, it's not possible to declare multiple `let` variables in the for-loop scope using separate `let` declarations, and therefore the rule allows `let i = 0, j = 2` in for-loop initializers because there is no equivalent with two `let` declarations. `let  j = 2; for (let i = 0; i < 0; i++) { ... }` declares `j` in the upper scope, so it's not the same as `for (let i = 0, j = 2; i < 0; i++) { ... }`

You can use [no-restricted-syntax](https://eslint.org/docs/rules/no-restricted-syntax) with `ForStatement > VariableDeclaration[declarations.length > 1].init` selector to disallow mutliple declarations in for-loop initializers:

```js
/*eslint no-restricted-syntax: ["error", "ForStatement > VariableDeclaration[declarations.length > 1].init"]*/

// allowed
let j = 2;
for(let i = 0; i < 0; i++) {
    console.log(i, j);
}

// disallowed
for(let i = 0, j = 2; i < 0; i++) {
    console.log(i, j);
}
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcmVzdHJpY3RlZC1zeW50YXg6IFtcImVycm9yXCIsIFwiRm9yU3RhdGVtZW50ID4gVmFyaWFibGVEZWNsYXJhdGlvbltkZWNsYXJhdGlvbnMubGVuZ3RoID4gMV0uaW5pdFwiXSovXG5cbi8vIGFsbG93ZWRcbmxldCBqID0gMjtcbmZvcihsZXQgaSA9IDA7IGkgPCAwOyBpKyspIHtcbiAgICBjb25zb2xlLmxvZyhpLCBqKTtcbn1cblxuLy8gZGlzYWxsb3dlZFxuZm9yKGxldCBpID0gMCwgaiA9IDI7IGkgPCAwOyBpKyspIHtcbiAgICBjb25zb2xlLmxvZyhpLCBqKTtcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

It is indeed debatable whether or not the same reasoning applies to `var` variables, but at this point when `var` variables are rarely used it doesn't seem that this case warrants revisiting the rule, especially since there's an easy workaround with `no-restricted-syntax`.
